runtime/src/message_processor.rs

```diff
@@ -183,18 +193,26 @@ impl MessageProcessor {
             .collect();
 
         self.process_instruction(message, instruction, executable_accounts, program_accounts)?;
-
         // Verify the instruction
-        for ((pre_program_id, pre_lamports, pre_data), post_account) in
-            pre_data.iter().zip(program_accounts.iter())
+        for ((pre_program_id, pre_lamports, pre_data), (i, post_account, is_credit_debit)) in
+            pre_data.iter().zip(
+                program_accounts
+                    .iter()
+                    .enumerate()
+                    .map(|(i, program_account)| (i, program_account, message.is_credit_debit(i))),
+            )
         {
             verify_instruction(
+                is_credit_debit,
                 &program_id,
                 pre_program_id,
                 *pre_lamports,
                 pre_data,
                 post_account,
             )?;
+            if !is_credit_debit {
+                *credits[i] += post_account.lamports - *pre_lamports;
```

Comment:
Probably need a test for a program that sets `lamports` below `pre_lamports`.  I'd expect a panic here if no `saturating_sub()`.

Reply:
`verify_instruction1 ensures that `post_account.lamports >= pre_lamports` for `!is_debitable` accounts, and errors out `execute_instruction` before reaching this line.
It's tested in `test_verify_instruction_credit_only` and `test_process_message_credit_only_handling`.

What am I missing?

Reply:
Must be me that's missing something